addons/account/wizard/account_payment_register.py

```diff
@@ -175,9 +175,7 @@ def _get_batch_journal(self, batch_result):
         payment_values = batch_result['payment_values']
         foreign_currency_id = payment_values['currency_id']
         partner_bank_id = payment_values['partner_bank_id']
-        company = batch_result['lines'].company_id
-        if len(company) > 1:
-            company = company._accessible_branches()[:1]
+        company = batch_result['lines'][0].company_id._accessible_branches()[:1]
```

Comment:
I think it should be something like, but it requires testing to register payments with invoices coming from different level of company hierarchy (one branch and it's parent, but not siblings)

```suggestion
        company = max(batch_result['lines'][0].company_id, key=lambda company: len(company.parent_ids))
```